### ISSUE TITLE
Ranker distant supervision details

### ISSUE THREAD

**Issue by chijames** [moutaigua8183/BookQA#4]
Hi,

I have several questions regarding the implementation of distant supervision of the ranker:
1. Which variant of BM25 did you use? Would be great if you could provide a pointer to the package you used :)
2. Did you perform any preprocessing/normalization to Qs and As when running the two BM25? For example, lowercase or stemming?
3. There are two answers per question, which one did you use to construct C_{Q+A}?
4. How many passages do the two weak BM25 retrievers retrieve? Namely, what are |C_{Q}| and |C_{Q+A}|?
5. What is the ratio (\sigma)of positives/negatives per question? 
5. For negative samples, are they coming from U-(C_{Q} \cap C_{Q+A}) or (C_{Q} \cup C_{Q+A}) - (C_{Q} \cap C_{Q+A}), where U is all the passages of a book.

Thanks!

**Comment by moutaigua8183** [comment]
To answer your questions:
1. Any BM25 should work, e.g., [this](https://pypi.org/project/rank-bm25/). I don't think they will make a significant difference.
2. we just simply convert everything to lowercase
3. We use both, i.e.,  Q + A1 + A2
4. 50
5. It is a hyperparameter, which is 8 in our case.  But you might need to try a few other numbers.
6. I don't quite understand your notations here. Literally, the intersection is the positive source; all others are the negative source.

Thanks!

**Comment by chijames** [comment]
Thanks for the reply! For 6, are they other "retrieved" passages, or all available passages in a book?

**Comment by moutaigua8183** [comment]
> Thanks for the reply! For 6, are they other "retrieved" passages, or all available passages in a book?

Both theoretically and practically, I don't think there are many differences. But I would recommand you picking negative samples of different types, such as the ones retrieved by ranker-A but not ranker-B,  the ones retrieved by ranker-B but not ranker-A, and the the ones retrieved by neither of them.

**Comment by chijames** [comment]
Sorry to reopen this! 
Is it possible to give the \alpha and \beta you used in the rouge-L filter? Apologies if they are mentioned somewhere in the paper and I miss them.
Thanks!

**Comment by chijames** [comment]
Also, I realize that it's not able to share the ranker data due to copyright issue, but can you give the basic statistics (mainly numbers) of the final generated positive and negative samples? I can use them to roughly verify my implementation.

**Comment by moutaigua8183** [comment]
> Sorry to reopen this!
> Is it possible to give the \alpha and \beta you used in the rouge-L filter? Apologies if they are mentioned somewhere in the paper and I miss them.
> Thanks!

\alpha = 0.4
\beta  = 0.8

**Comment by moutaigua8183** [comment]
> Also, I realize that it's not able to share the ranker data due to copyright issue, but can you give the basic statistics (mainly numbers) of the final generated positive and negative samples? I can use them to roughly verify my implementation.

I don't have the exact numbers now.  If I remember correctly, there are ~150k positive samples and ~1.2m negative samples for the ranker before Rouge-L filtering;  and they become 10k+ and ~100k+ respectively after filtering.

**Comment by abhinavkashyap** [comment]
Hi :) I came across this repository and I was wondering what results to expect for training of the ranker 
I assume that the example in the ranker section shows the default hyperparams. 
What is the Accuracy and F1 score for the bert-base-uncased rankers on data created using distant supervision. 
Any input would be helpful :) Thanks

**Comment by moutaigua8183** [comment]
> Hi :) I came across this repository and I was wondering what results to expect for training of the ranker I assume that the example in the ranker section shows the default hyperparams. What is the Accuracy and F1 score for the bert-base-uncased rankers on data created using distant supervision. Any input would be helpful :) Thanks

Hi, thanks for your interests!  The actual performance will depend on the actual data, the user case and maybe the base ranker too. The setting in our paper (NarrativeQA evidence retrieval) is a challenging task and we achieve ~50 in rolling Rouge-L.  so your task (or data)  is less challenging, I am expecting a number higher than 50.

**Comment by abhinavkashyap** [comment]
Thanks for the response :) I am trying to replicate on the same dataset. I see from the paper that the rolling Rouge-L is 50. But any F1 score for the classification using bert-base-uncased?

**Comment by moutaigua8183** [comment]
> Thanks for the response :) I am trying to replicate on the same dataset. I see from the paper that the rolling Rouge-L is 50. But any F1 score for the classification using bert-base-uncased?

I am sorry but we don't compute the F1 scores for rankers. We only computed rolling Rouge-L and exact match.  I am expecting the F1 score should be much lower than 50,   most likely in a range between 30 and 40 I guess.
